Mathlib/MeasureTheory/Measure/SeparableMeasure.lean

```diff
@@ -0,0 +1,494 @@
+              exact mul_le_mul_of_nonneg_left
+                ((div_le_one (by linarith [μs_pow_nonneg])).2 (by linarith))
+                (by linarith [ε_pos])
+      · rw [← dist_eq_norm, dist_indicatorConstLp_eq_norm,
+          norm_indicatorConstLp p_ne_zero p_ne_top.elim]
+        have : (μ (s ∆ t)).toReal ^ (1 / p.toReal) ≤ ε / (3 * ( 1 + ‖b‖)) := by
+          rw [← rpow_le_rpow_iff (rpow_nonneg toReal_nonneg _)
+              (div_nonneg (le_of_lt ε_pos) (by linarith [norm_nonneg b]))
+              (toReal_pos p_ne_zero p_ne_top.elim), one_div,
+            rpow_inv_rpow toReal_nonneg (toReal_ne_zero.2 ⟨p_ne_zero, p_ne_top.elim⟩),
+            ← toReal_ofReal <| le_of_lt approx_s_pos]
+          exact toReal_mono ofReal_ne_top (le_of_lt hμst)
+        calc
+          ‖b‖ * (μ (s ∆ t)).toReal ^ ( 1 / p.toReal)
+            ≤ ‖b‖ * (ε / (3 * ( 1 + ‖b‖))) := mul_le_mul_of_nonneg_left this (norm_nonneg ..)
+          _ ≤ ε / 3 := by
+              rw [← mul_one (ε / 3), div_mul_eq_div_mul_one_div, ← mul_assoc, mul_comm ‖b‖,
+                mul_assoc, mul_one_div]
+              exact mul_le_mul_of_nonneg_left
+                ((div_le_one (by linarith [norm_nonneg b])).2 (by linarith))
+                (by linarith [ε_pos])
+    · -- Now we have to show that the closure of `D` is closed by sum. Let `f` and `g` be two
+      -- functions in `Lᵖ` which are also in the closure of `D`.
+      rintro f g hf hg - f_mem g_mem
+      rw [SeminormedAddCommGroup.mem_closure_iff] at *
+      intro ε ε_pos
+      -- For `ε > 0`, there exists `bf, bg ∈ D` such that `‖f - bf‖ₚ < ε/2` and `‖g - bg‖ₚ < ε/2`.
+      rcases f_mem (ε / 2) (by linarith [ε_pos]) with ⟨bf, ⟨nf, df, sf, bf_eq⟩, hbf⟩
+      rcases g_mem (ε / 2) (by linarith [ε_pos]) with ⟨bg, ⟨ng, dg, sg, bg_eq⟩, hbg⟩
+      -- It is obvious that `D` is closed by sum, it suffices to concatenate the family of
+      -- elements of `u` and the family of elements of `𝒜₀`.
+      let d := fun i : Fin (nf + ng) ↦ if h : i < nf
+        then df (Fin.castLT i h)
+        else dg (Fin.subNat nf (Fin.cast (Nat.add_comm ..) i) (le_of_not_gt h))
+      let s := fun i : Fin (nf + ng) ↦ if h : i < nf
+        then sf (Fin.castLT i h)
+        else sg (Fin.subNat nf (Fin.cast (Nat.add_comm ..) i) (le_of_not_gt h))
+      -- So we can use `bf + bg`.
+      refine ⟨bf + bg, ⟨nf + ng, d, s, ?_⟩, ?_⟩
+      · simp [key, d, s, Fin.sum_univ_add, bf_eq, bg_eq]
+      · -- We have
+        -- `‖f + g - (bf + bg)‖ₚ`
+        --   `≤ ‖f - bf‖ₚ + ‖g - bg‖ₚ`
+        --   `< ε/2 + ε/2 = ε`.
+        calc
+          ‖Memℒp.toLp f hf + Memℒp.toLp g hg - (bf + bg)‖
+            = ‖(Memℒp.toLp f hf) - bf + ((Memℒp.toLp g hg) - bg)‖ := by congr; abel
+          _ ≤ ‖(Memℒp.toLp f hf) - bf‖ + ‖(Memℒp.toLp g hg) - bg‖ := norm_add_le ..
+          _ < ε := by linarith [hbf, hbg]
+    · -- Obviously the closure of `D` is closed.
+      exact isClosed_closure
```

Comment:
When a goal is trivial like that, you can deal with it right away in the calling line, i.e., use
```
    refine Lp.induction p_ne_top.elim (P := fun f ↦ f ∈ closure D) ?_ ?_ isClosed_closure
```
above.